### PR TITLE
ECAL : Tool to create sqlite files for DB

### PR DESCRIPTION
**Pull request by depasse**
#### PR description:

ECAL : for appending EcalTimeCalibConstants tags with new payloads : tool to create sqlite files from raw xxx.dat or xxx.xml files.

#### PR validation:

Tested (cmsRun updateTimeCalib.py) in CMSSW_13 and 14_0_5 with xxx.dat and xxx.xml source files  : same payloads generated (hashes).
Comments about const suggestion in #PR 44713 don't work, after many attempts.  Still compilation errors.  N.B. All other ...Handler.cc of CondTools/Ecal/src have the same structure , without const.

### CODE REVIEW

**Comment by perrotta** [comment]
@depasse the issue of the const'ness of the class members is indeed rather minor for this kind of code, even though it could allow making the code safer and in principle it must be preferred.
In any case, I was curious to see why it wouldn't be possible to make those unmodified members constant, as they should. Therefore, I made an attempt following what was already proposed in the previous version of this PR and, as expected, everything built correctly as such. I've coded this update and put in github in the second commit of perrotta:cleanupEcalTimeCalibHandler
In that commit I also took the opportunity to clean up the code a bit, and remove a bunch of unneeded references. I did not check whether there is any different result with respect to what can be obtained with your code (but I don't see why it should), and I would let you check it. My suggestion would be to cherry-pick that commit here, if you agree with it, and clean up the code here with it.

**Comment by depasse** [comment]
Could you remind me the command  ?  I tried  
git cherry-pick https://github.com/perrotta/cmssw/tree/cleanupEcalTimeCalibHandler
or with the hash but it is not the right method.

**Comment by perrotta** [comment]
does
```
git cherry-pick 70c38f483930b81fe98231457a8a755173d0903d
```
work? (Where the tag hash is taken from https://github.com/perrotta/cmssw/tree/cleanupEcalTimeCalibHandler)

**Comment by depasse** [comment]
I had tried too, but with the same result :  $ git cherry-pick 70c38f483930b81fe98231457a8a755173d0903d
fatal: bad object 70c38f483930b81fe98231457a8a755173d0903d.

I do this command from /afs/cern.ch/user/d/depasse/WORK/CMSSW_14_1_0_pre2/src where CondTools/Ecal is present

**Comment by depasse** [comment]
Otherwise I simply copy/paste the .h and .cc (the only (?) files you adapted)

**Comment by depasse** [comment]
I have closed PR #44713. , applied  scram build code-checks, then  scram build code-format.

**Comment by perrotta** [comment]
please test

File: CondTools/Ecal/src/EcalTimeCalibHandler.cc [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      m_firstRun(static_cast<unsigned int>(atoi(ps.getParameter<std::string>("firstRun").c_str()))),
+      m_file_name(ps.getParameter<std::string>("fileName")),
+      m_file_type(ps.getParameter<std::string>("type"))  // xml/txt
+{
+  edm::LogInfo("EcalTimeCalib Source handler constructor\n");
+}
+
+void popcon::EcalTimeCalibHandler::getNewObjects() {
+  std::ostringstream ss;
+  ss << "ECAL ";
```
Comment (oljemark) [inline CondTools/Ecal/src/EcalTimeCalibHandler.cc:19]: Hello. Is this string stream used somewhere in the function?

**Review by perrotta — COMMENTED** [review]
Some further possible cleaning

**Comment by depasse** [comment]
Problem with the added #include
Compiling  src/CondTools/Ecal/src/EcalTimeCalibHandler.cc
src/CondTools/Ecal/src/EcalTimeCalibHandler.cc:5:10: fatal error: cstdef: No such file or directory
    5 | #include <cstdef>
      |          ^~~~~~~~

First lines  of  EcalTimeCalibHandler.cc :
#include "CondTools/Ecal/interface/EcalTimeCalibHandler.h"
#include "DataFormats/EcalDetId/interface/EEDetId.h"
#include "DataFormats/EcalDetId/interface/EBDetId.h"
#include "FWCore/MessageLogger/interface/MessageLogger.h"
#include <cstdef>
#include <cstdio>
#include <fstream>
#include <sstream>
#include <utility>

const Int_t kEBChannels = 61200, kEEChannels = 14648;

**Comment by thomreis** [comment]
> does
> 
> ```
> git cherry-pick 70c38f483930b81fe98231457a8a755173d0903d
> ```
> 
> work?

I think in order to get the `cherry-pick` command to work one would have to first add `perrotta` as a remote and `fetch`.|
```bash
git remote add perrotta git@github.com:perrotta/cmssw.git
git fetch perrotta
git cherry-pick 70c38f483930b81fe98231457a8a755173d0903d
```

**Comment by depasse** [comment]
Found your typo  : it was cstddef  and not  #include <cstdef>

**Comment by thomreis** [comment]
type ecal

**Comment by thomreis** [comment]
please test

**Comment by depasse** [comment]
Yes, iI had verified this in 14_1_0_pre2  (present PR) before git push, :  still the same hashes produced (as with CMSSW_14_1_X_2024-04-10-2300, previous PR)

**Comment by perrotta** [comment]
+db
- See https://github.com/cms-sw/cmssw/pull/44794#issuecomment-2074414576

**Comment by antoniovilela** [comment]
@depasse 
Now please squash the commits [1]. If the squash is done correctly, the PR will not need to be re-signed.
Thanks.
[1]
https://cms-sw.github.io/faq.html#how-do-i-collapse-multiple-commits-into-one
https://git-scm.com/book/en/Git-Tools-Rewriting-History
N.B.
Always create a backup branch.

**Comment by depasse** [comment]
I never made this.  To squash correctly, can you give me the right commands in the right order. ?  Are these enough ?
git  reset --soft  e3fa0a1ab20b35c70e6d3dfd3274be656dff7ff8
git commit --amend
(git rebase ?)
How do you create a backup branch ? To do before the squash ?

**Comment by perrotta** [comment]
> I never made this. To squash correctly, can you give me the right commands in the right order. ? Are these enough ? git reset --soft [e3fa0a1](https://github.com/cms-sw/cmssw/commit/e3fa0a1ab20b35c70e6d3dfd3274be656dff7ff8) git commit --amend (git rebase ?) How do you create a backup branch ? To do before the squash ?

@depasse, I would do the following:

```
git log
```
(to get the list of commits before your owns)
```
git reset --soft [commit]
```
(reset to the last commit **before** your first one, as from the previous list)
```
git commit -m "ECAL: Tool to create sqlite files for DB"
```
(I imagine you want to keep the same comment as of now)
```
git push -f my-cmssw ToolEcalTimeCalibConstants
```

As a backup, probably in your case the easiest way is to copy somewhere the four modified files before attempting the squash

**Comment by antoniovilela** [comment]
> > I never made this. To squash correctly, can you give me the right commands in the right order. ? Are these enough ? git reset --soft [e3fa0a1](https://github.com/cms-sw/cmssw/commit/e3fa0a1ab20b35c70e6d3dfd3274be656dff7ff8) git commit --amend (git rebase ?) How do you create a backup branch ? To do before the squash ?
> 
> @depasse, I would do the following:
> 
> ```
> git log
> ```
> 
> (to get the list of commits before your owns)
> 
> ```
> git reset --soft [commit]
> ```
> 
> (reset to the last commit **before** your first one, as from the previous list)
> 
> ```
> git commit -m "ECAL: Tool to create sqlite files for DB"
> ```
> 
> (I imagine you want to keep the same comment as of now)
> 
> ```
> git push -f my-cmssw ToolEcalTimeCalibConstants
> ```
> 
> As a backup, probably in your case the easiest way is to copy somewhere the four modified files before attempting the squash

You could create a backup copy (for example ToolEcalTimeCalibConstants_backup), from the current branch before squashing. Like this you can recover if needed by force-pushing the backup branch in place of the PR branch.

**Comment by antoniovilela** [comment]
+1

**Pull request merged**